junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala

```diff
@@ -0,0 +1,447 @@
+          case pl if pl.getClass.getName == "org.scalajs.core.compiler.ScalaJSPlugin" => pl
+        }.get.asInstanceOf[ScalaJSPlugin]
+      }
+
+      override def transform(tree: Tree): Tree = tree match {
+        case tree: PackageDef =>
+          def classHasJUnitAnnotations(clazz: ClassDef): Boolean = {
+            clazz.impl.body.exists {
+              case ddef: DefDef => hasAnnotation[org.junit.Test](ddef)
+              case _ => false
+            }
+          }
+          val newStats = tree.stats.groupBy {
+            case clDef: ClassDef => clDef.name
+            case _ => "not a class def"
+          }.iterator.flatMap {
+            case ("not a class def", xs) => xs
+            case (className, x :: xs) =>
+              val clDef1 = x.asInstanceOf[ClassDef]
+              val clDef2Option = xs match {
+                case Nil => None
+                case y :: ys =>
+                  if (ys.isEmpty) Some(y.asInstanceOf[ClassDef])
+                  else None
+              }
+              val (clDefOption, mlDefOption) = {
+                if (clDef1.mods.hasFlag(Flags.MODULE))
+                  (clDef2Option, Some(clDef1))
+                else (Some(clDef1), clDef2Option)
+              }
+
+              if ((clDefOption.isDefined && classHasJUnitAnnotations(clDefOption.get)) ||
+                (mlDefOption.isDefined && classHasJUnitAnnotations(mlDefOption.get))) {
+                val transformedClDef = clDefOption match {
+                  case Some(clDef) => transformTestClass(clDef)
+
+                  case None =>
+                    throw new ClassNotFoundException(
+                      mlDefOption.get.name.toString)
+                }
+                val hookClass = mkHookClass(transformedClDef, mlDefOption)
+                hookClass :: transformedClDef :: mlDefOption.toList
+              }
+              else x :: xs
+
+            case (_, Nil) => Nil
+          }
+
+          val newPackage = tree.copy(stats = newStats.toList)
+          newPackage.setSymbol(tree.symbol)
+          newPackage
+
+        case _ =>
+          super.transform(tree)
+      }
+
+      def getLocalMethodSymbol(clazz: ClassDef): Symbol = {
+        clazz.impl.body.collectFirst {
+          case dDef: DefDef if hasAnnotation[org.junit.Test](dDef) => dDef
+        }.get.symbol
+      }
+
+      def transformTestClass(clazz: ClassDef): ClassDef = {
+        inform("Scala.js JUnit plugin: transforming " + clazz.name)
+        val invokeJUnitMethodDef = {
+          val annotatedMethods = jUnitAnnotatedMethods(clazz)
+          val localMethodSymbol = getLocalMethodSymbol(clazz)
+          mkInvokeJUnitMethodDef(annotatedMethods, localMethodSymbol)
+        }
+
+        val newBody = invokeJUnitMethodDef :: clazz.impl.body
+        val newParents = clazz.impl.parents :::
+          TypeTree(typeOf[ScalaJSJUnitTest]) :: Nil
+        val newImpl = clazz.impl.copy(parents = newParents, body = newBody)
```

Comment:
Use `treeCopy` (I guess you can Ctrl+F for `.copy` and change most of them to use `treeCopy`)